cinema/serializers.py

```diff
@@ -1 +1,85 @@
-# write serializers here
+from rest_framework import serializers
+
+from cinema.models import Movie, Genre, Actor, MovieSession, CinemaHall
+
+
+class ActorsSerializer(serializers.ModelSerializer):
+    full_name = serializers.SerializerMethodField()
+
+    def get_full_name(self, actor):
+        return actor.first_name + " " + actor.last_name
+
+    class Meta:
+        model = Actor
+        fields = ("id", "first_name", "last_name", "full_name")
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = ("id", "name")
+
+
+class CinemaHallSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CinemaHall
+        fields = ("id", "name", "rows", "seats_in_row", "capacity")
+
+
+class MovieSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Movie
+        fields = ("id", "title", "description", "duration", "genres", "actors")
+
+
+class MovieListSerializer(MovieSerializer):
+    genres = serializers.SlugRelatedField(
+        many=True,
+        read_only=True,
+        slug_field="name",
+    )
+    actors = serializers.SerializerMethodField()
+
+    def get_actors(self, movie: Movie) -> list[str]:
+        return [
+            f"{actor.first_name} {actor.last_name}"
+            for actor in movie.actors.all()
+        ]
+
+
+class MovieRetrieverSerializer(MovieSerializer):
+    genres = GenreSerializer(many=True)
+    actors = ActorsSerializer(many=True)
```

Comment:
I recommend implement this fields like read_only, because you don't need to provide genres and actors data for update movies